File: TiaCodegen/Commands/Functions/Base/SystemFunctionBlockCall.cs
using TiaCodegen.Interfaces;

namespace TiaCodegen.Commands.Functions.Base
{
    public class SystemFunctionBlockCall : FunctionBlockCall
    {
        public SystemFunctionBlockCall(string functionName, string instanceName, IOperationOrSignal eno = null) : base(functionName, instanceName, eno)
        {
        }

        public string TemplateValueName { get; set; }
        public string TemplateValueType { get; set; }
        public string TemplateValue { get; set; }
    }
}

File: TiaCodegen-ts/src/Commands/Functions/Base/SystemFunctionBlockCall.ts
import { FunctionBlockCall, FunctionBlockCallOptions } from './FunctionBlockCall.js';

export interface SystemFunctionBlockCallOptions extends FunctionBlockCallOptions {}

export class SystemFunctionBlockCall extends FunctionBlockCall {
    templateValueName: string | null = null;
    templateValueType: string | null = null;
    templateValue: string | null = null;

    constructor(options: SystemFunctionBlockCallOptions) {
        super(options);
    }
}
